androidshared/src/main/java/org/odk/collect/androidshared/ui/FixMismatchThemeProblemsActivity.kt

```diff
@@ -0,0 +1,43 @@
+package org.odk.collect.androidshared.ui
+
+import android.os.Build
+import android.os.Bundle
+import android.webkit.WebView
+import androidx.appcompat.app.AppCompatActivity
+
+/**
+ * When the app's theme doesn't match the system, the first [WebView] created will cause the host
+ * Activity to be recreated with incorrect resource loading (the system ones instead of the custom
+ * set dark/light resources). This doesn't happen for subsequent [WebView] creations however.
+ *
+ * This Activity can be started at launch to be "sacrificed" so no other Activity has to deal
+ * with the problem.
+ *
+ * See [this issue](https://issuetracker.google.com/issues/37124582) for more details.
+ */
+class FixMismatchThemeProblemsActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (alreadyLaunched) {
+            finish()
+            return
+        }
+
+        if (Build.VERSION.SDK_INT >= 24) {
+            try {
+                WebView(this)
```

Comment:
In the description, you have briefly mentioned the issue that we had with webviews. It would be good to let the QA team know what steps were needed to reproduce it so that they can confirm there is no regression. I belive it's this https://github.com/getodk/collect/pull/4803#issuecomment-907092440 if so please add it to the description.

Reply:
Done!